Wikipedia/Code/NSMutableAttributedString+Highlight.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+extension NSMutableAttributedString {
+
+    /// Highlight occurrences of substring
+    /// - Parameters:
+    ///   - text: the text to highlight
+    ///   - backgroundColor: the color to use in the background for matches
+    ///   - foregroundColor: the color to use as the foreground for matches
+    ///   - targetRange: if nil, highlight all matches in `self`. If non-nil, only highlight matches in the target `NSRange`
+    /// - Returns: the attributed string with highlight attributes added
+    func highlight(_ text: String?, backgroundColor: UIColor, foregroundColor: UIColor? = nil, targetRange: NSRange? = nil) -> NSMutableAttributedString {
+        guard let text = text else { return self }
+        
+        let attributedString = self
+
+        var ranges = (string as NSString).ranges(of: text)
```

Comment:
I agree there are more performant ways to construct this. The reason I went with this particular approach though is to prevent misuse of the target range. I didn't want to allow a engineer to accidentally provide a `NSRange` that is potentially out of range for the string, or just provide an arbitrary range that doesn't actually contain the search text.

But I acknowledge this may be poor or confusing method design on my part trying to consolidate this highlight code in one place rather than potentially breaking it up more. I don't mind changing this or breaking it up somehow if you'd like as long as we're still providing some of that in-built range safety.

What do you think?

Reply:
@staykids Ah! That makes sense, I didn't think about potential misuse like that. I think it's fine to leave as-is then, and revisit if we notice any performance problems.